bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -1087,7 +1089,7 @@ protected synchronized RuleStatusInfo getRuleStatusInfo(String rUID) {
         return statusMap.get(rUID);
     }
 
-    protected synchronized String getUniqueId() {
+    protected synchronized @Nonnull String getUniqueId() {
```

Comment:
> "because I know"

Now that is really helpful, is it? If somebody refactors the code so that it suddenly returns null, all your code will break without noticing. If somebody adds a functionality that uses your method, he will add null checks, because he does not know that it will always be non-null. For both situations, the annotation is very helpful.